src/Integration/Monolog/CloudWatch/src/CloudWatchLogsHandler.php

```diff
@@ -0,0 +1,295 @@
+     */
+    private $group;
+
+    /**
+     * @var string
+     */
+    private $stream;
+
+    /**
+     * @var bool
+     */
+    private $initialized = false;
+
+    /**
+     * @var string|null
+     */
+    private $sequenceToken;
+
+    /**
+     * @var int
+     */
+    private $batchSize;
+
+    /**
+     * @var array
+     */
+    private $buffer = [];
+
+    /**
+     * @var int
+     */
+    private $currentDataAmount = 0;
+
+    /**
+     * @var int
+     */
+    private $remainingRequests = self::RPS_LIMIT;
+
+    /**
+     * @var \DateTimeImmutable
+     */
+    private $savedTime;
+
+    /**
+     * @param CloudWatchLogsClient $client
+     *
+     *  Log group names must be unique within a region for an AWS account.
+     *  Log group names can be between 1 and 512 characters long.
+     *  Log group names consist of the following characters: a-z, A-Z, 0-9, '_' (underscore), '-' (hyphen),
+     * '/' (forward slash), and '.' (period).
+     * @param string $group
+     *
+     *  Log stream names must be unique within the log group.
+     *  Log stream names can be between 1 and 512 characters long.
+     *  The ':' (colon) and '*' (asterisk) characters are not allowed.
+     * @param string $stream
+     * @param int    $batchSize
+     * @param int    $level
+     * @param bool   $bubble
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function __construct(
+        CloudWatchLogsClient $client,
+        $group,
+        $stream,
+        $batchSize = 10000,
+        $level = Logger::DEBUG,
+        $bubble = true
+    ) {
+        if ($batchSize > 10000) {
+            throw new \InvalidArgumentException('Batch size can not be greater than 10000');
+        }
+
+        $this->client = $client;
+        $this->group = $group;
+        $this->stream = $stream;
+        $this->batchSize = $batchSize;
+
+        parent::__construct($level, $bubble);
+
+        $this->savedTime = new \DateTimeImmutable();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function close(): void
+    {
+        $this->flushBuffer();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getDefaultFormatter(): FormatterInterface
+    {
+        return new LineFormatter('%channel%: %level_name%: %message% %context% %extra%', null, false, true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function write(array $record): void
+    {
+        $records = $this->formatRecords($record);
+
+        foreach ($records as $record) {
+            if ($this->currentDataAmount + $this->getMessageSize($record) >= self::DATA_AMOUNT_LIMIT) {
+                $this->flushBuffer();
+            }
+
+            $this->currentDataAmount += $this->getMessageSize($record);
+
+            $this->buffer[] = $record;
+
+            if (\count($this->buffer) >= $this->batchSize) {
+                $this->flushBuffer();
+            }
+        }
+    }
+
+    private function checkThrottle(): void
```

Comment:
Im not sure why this is needed. Why do we sleep 1 second? Why is 1000ms such a magic number... 

I think we should rework this in the next PR.

Reply:
5 RPS comes from the official documentation: https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html
I have copied that in the phpdoc

I agree the throttling mechanism is very basic. The original author allows 5 requests within the same second, instead of waiting 1 sec after the 5th request. In both cases, I would not choose any other value than 1000ms. 

Yet, this doesn't throttle requests coming from different processes, but it's still better than nothing